internal/storage/storage.go

```diff
@@ -100,6 +103,36 @@ func (m *Memory) Ping(_ context.Context) error {
 	return errors.New("there is no connection to DB")
 }
 
+func (m *Memory) DeleteForUser(ctx context.Context, _ *sync.WaitGroup, inputCh chan middleware.ItemDelete) {
+
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case item := <-inputCh:
```

Comment:
```suggestion
		case item, opened := <-inputCh:
			if !opened {
				return
			}
```
Если уж канал в какой-то момент закрывается, не стоит пытаться из него читать. А на закрытый канал case от select будет срабатывать всегда.

Reply:
спасибо, исправлено